care/emr/api/viewsets/patient.py

```
@@ -394,15 +402,35 @@ def get_tokens(self, request, *args, **kwargs):
         self.authorize_update({}, patient)
         request_config = get_object_or_404(
             PatientIdentifierConfig, external_id=request_data.config
         )
+
+        facility = self.get_serializer_list_context().get("facility")
+
+        # Ensure identifier config belongs to the patient's facility context
+        if request_config.facility and facility and request_config.facility.id != facility.id:
+            raise PermissionDenied(
+                "Identifier configuration does not belong to the patient's facility"
+            )
```

Ruff (0.15.4)

[warning] Avoid specifying long messages outside the exception class

(TRY003)